### ISSUE TITLE
#macro argument should not be outside of macro context

### ISSUE THREAD

**Issue by matter123** [jeff-hykin/cpp-textmate-grammar#364]
By having `variable.other.macro.argument` be outside of the `macro_context`, An annoying result happens. As you type a preprocessor directive, the text is colored as a variable, when you finish typing the directive suddenly, the word is treated as a keyword.

Outside of a macro definition, any line that starts with `#` should have the scopes `keyword.control.directive` or `source.cpp` depending on the validity of the word that follows.

Optionally if the directive is not a recognized directive and It is not the only thing on the line (this is a probable indicator that the directive is being typed), then consider the directive to be illegal, with a scope of `invalid.unknown.keyword.control.directive`

![peekW0PH6Z](https://user-images.githubusercontent.com/714007/63472653-a6a34f80-c427-11e9-9c52-be650edda43a.gif)

**Comment by jeff-hykin** [comment]
Yeah once we are able to limit the arguments to the #define scope via some feature on the bailout, then I think thats a good solution
